src/main/java/com/cleanroommc/groovyscript/compat/mods/additionalenchantedminer/WorkBenchPlus.java

```diff
@@ -0,0 +1,100 @@
+package com.cleanroommc.groovyscript.compat.mods.additionalenchantedminer;
+
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.documentation.annotations.*;
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import com.yogpc.qp.recipe.WorkbenchRecipe;
+import com.yogpc.qp.tile.ItemDamage;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+import org.jetbrains.annotations.Nullable;
+import scala.Option;
+import scala.collection.JavaConversions;
+import scala.collection.Map;
+
+@RegistryDescription
+public class WorkBenchPlus extends VirtualizedRegistry<WorkbenchPlusRecipe> {
+
+    @Override
+    public void onReload() {
+        removeScripted().forEach(recipe -> WorkbenchPlusRecipe.removeById(recipe.getLocation().toString()));
+        restoreFromBackup().forEach(WorkbenchPlusRecipe::addRecipe);
+    }
+
+    @MethodDescription(example = @Example("item('quarryplus:quarry')"))
+    public Boolean removeByOutput(ItemStack output) {
+        ItemDamage itemDamage = ItemDamage.apply(output);
+        Map<ResourceLocation, WorkbenchRecipe> recipeMap = WorkbenchRecipe.getRecipeMap();
+        Iterable<WorkbenchRecipe> iterable  = JavaConversions.asJavaIterable(recipeMap.values());
+        iterable.forEach(recipe -> {
+            if (recipe.key().equals(itemDamage)) {
+                addBackup(new WorkbenchPlusRecipe(recipe.inputs(), recipe.getOutput(), recipe.energy(), recipe.location()));
+            }
+        });
+        return WorkbenchPlusRecipe.removeByOutput(output);
+    }
+
+    public Boolean removeById(String id) {
```

Comment:
should this be used by people using GroovyScript be calling this method? if so, use `@MethodDescription` and state what it does. otherwise, private or `@GroovyBlacklist`. same with `removeAll`.